agent/testflinger_agent/agent.py

```diff
@@ -335,3 +338,12 @@ def retry_old_results(self):
             except TFServerError:
                 # Problems still, better luck next time?
                 pass
+
+    def restart_signal_handler(self, _, __):
+        """
+        If we receive the restart signal, tell the agent to restart safely when
+        it is not running a job
+        """
+        logger.info("Triggering agent restart")
```

Comment:
If I understand this correctly, the agent isn't restarted immediately. It's restarted whenever `check_restart` gets called after this restart file was created. If so would it make sense to log `Marked agent for restart` instead?

Reply:
That's why I said "Triggering" instead of just "restarting agent" but if that's more clear to call it "Marked" then I'm happy to do that. Changing it now, will push an update in just a minute.